dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.PushNotifications
+{
+    // Event args for the Push payload.
+    runtimeclass PushNotificationReceivedEventArgs
+    {
+        // Initialize using the IBackgroundInstance: used specifically for the Background Activation scenario
+        static PushNotificationReceivedEventArgs CreateFromBackgroundTaskInstance(Windows.ApplicationModel.Background.IBackgroundTaskInstance backgroundTask);
+
+// Initialize using the PushNotificationEventArgs from Windows: used specifically for in-memory event handling when app is already in foreground
+static PushNotificationReceivedEventArgs CreateFromPushNotificationReceivedEventArgs(Windows.Networking.PushNotifications.PushNotificationReceivedEventArgs args);
+
+// The Push payload
+byte[] Payload { get; };
+
+// Gets a deferral to run under specific modes like low power mode
+Windows.ApplicationModel.Background.BackgroundTaskDeferral GetDeferral();
+
+// Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+// Set to true to prevent proceeding launch due to Background Activation: false by default
+Boolean Handled;
+};
+
+[flags]
+enum PushNotificationRegistrationKind
```

Comment:
Flags enums cannot be a `Kind`. A `Kind` is a list of possibilities. These are more like `Options`.

And what happens if they don't pass any options? Does nothing get registered? Is that okay?

Reply:
As of right now, nothing gets registered if no options are passed. This is an invalid case so I will a option check in RegisterActivator. I can rename kind -> options as well.

Reply:
Since it is a enum, how could someone pass in no options?